File: shoppie_bff/config/app_config.go
package config

import "os"

type AppConfig struct {
	UserServiceBaseUrl    string
	ProductServiceBaseUrl string
	OrderServiceBaseUrl   string
	CartServiceBaseUrl    string
	ServerPort            string
}

func NewAppConfig() *AppConfig {
	return &AppConfig{
		UserServiceBaseUrl:    os.Getenv("USER_SERVICE_BASE_URL"),
		ProductServiceBaseUrl: os.Getenv("PRODUCT_SERVICE_BASE_URL"),
		OrderServiceBaseUrl:   os.Getenv("ORDER_SERVICE_BASE_URL"),
		CartServiceBaseUrl:    os.Getenv("CART_SERVICE_BASE_URL"),
		ServerPort:            os.Getenv("SERVER_PORT"),
	}
}

File: frontend/shoppie_admin/lib/app/app_config.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';

class AppConfig {
  static String userServiceBaseUrl = dotenv.env["USER_SERVICE_BASE_URL"] ?? "";

  static String productServiceBaseUrl =
      dotenv.env["PRODUCT_SERVICE_BASE_URL"] ?? "";

  static String cartServiceBaseUrl = dotenv.env["CART_SERVICE_BASE_URL"] ?? "";

  static String orderServiceBaseUrl =
      dotenv.env["ORDER_SERVICE_BASE_URL"] ?? "";

  static String jwtSecret = dotenv.env["JWT_SECRET"] ?? "";

  static String baseUrl = dotenv.env["BASE_URL"] ?? "";
}
